src/picologging/logger.cxx

```diff
@@ -48,12 +48,12 @@ PyObject* Logger_new(PyTypeObject* type, PyObject* args, PyObject* kwds)
         
         self->_fallback_handler = (StreamHandler*)PyObject_CallFunctionObjArgs((PyObject *)&StreamHandlerType, NULL);
         if (self->_fallback_handler == nullptr){
-            Py_DECREF(self->parent);
-            Py_DECREF(self->handlers);
-            Py_DECREF(self->manager);
+            Py_XDECREF(self->name);
```

Comment:
Why doesn't name also have to get DECREF'd in the if statement on line 41? (Since it gets INCREF'd before it)